easybuild/framework/extension.py

```diff
@@ -138,6 +138,10 @@ def __init__(self, mself, ext, extra_params=None):
                 self.log.debug("Skipping unknown custom easyconfig parameter '%s' for extension %s/%s: %s",
                                key, name, version, value)
 
+        if self.cfg['maxparallel'] and self.cfg['parallel'] > self.cfg['maxparallel']:
+            self.cfg['parallel'] = min(self.cfg['parallel'], self.cfg['maxparallel'])
```

Comment:
No need to compare again
```suggestion
            self.cfg['parallel'] = self.cfg['maxparallel']
```